api/storage_service.cc

```diff
@@ -785,7 +786,8 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         }
 
         auto& compaction_module = db.local().get_compaction_manager().get_task_manager_module();
-        auto task = co_await compaction_module.make_and_start_task<cleanup_keyspace_compaction_task_impl>({}, std::move(keyspace), db, table_infos);
+        auto task = co_await compaction_module.make_and_start_task<cleanup_keyspace_compaction_task_impl>(
+            {}, std::move(keyspace), db, table_infos, flush_mode::compacted_tables);
```

Comment:
why not `flush_mode::all_tables`?

Also, I think the patch title should be "flush all tables before cleanup"

Reply:
https://github.com/scylladb/scylladb/blob/cb5453d53452500f7448ef223e721a4244abd761/compaction/task_manager_module.hh#L56

but in this case, we let the task to perform the flush for the tables by itself. so i think `flush_mode::all_tables` does not apply here.

Reply:
thank you! indeed. changed.